src/features/sendEmail/SendEmail.tsx

```diff
@@ -0,0 +1,124 @@
+import {
+  Alert,
+  Box,
+  Button,
+  Snackbar,
+  TextField,
+  TextareaAutosize,
+  Typography,
+} from '@mui/material';
+import './SendEmail.css';
+import { useParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { useSendEmailMutation } from '../../app/appSlice';
+
+export default function SendEmail() {
+  const { receiverAddress, protectedDataAddress } = useParams();
+
+  //RTK Mutation hook
+  const [sendEmail, result] = useSendEmailMutation();
+
+  //for textarea
+  const [value, setValue] = useState('');
+  //limited to 4096 by the SMS
+  const [charactersRemainingMessage, setCharactersRemainingMessage] =
+    useState(4096);
+
+  //for name et dataType
+  const [messageSubject, setMessageSubject] = useState('');
+  //limited to 78 by the SMS
+  const [charactersRemainingSubject, setCharactersRemainingSubject] =
+    useState(78);
```

Comment:
do not add to the state something that is derived state

```ts
const charactersRemainingSubject = 78 - messageSubject.length;
```

Reply:
Commit: 44edd3cf75d5f0fe9cdeb3dae87fa32d41a0a8b1